### PR TITLE
Print the unrecognized repository rule type

### PR DESCRIPTION
**Pull request by tomfeigin**
When failing to match the type of a repository rule, print the unmatched name.
Previously only the type of the unmatched name was printed, which is always 'string'. This makes it hard to troubleshoot unmarshalling failures.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @tomfeigin !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by liaodaniel — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @liaodaniel !
Merging.

**Pull request merged**
